llvm/lib/Target/AMDGPU/AMDGPUSubtarget.cpp

```diff
@@ -191,17 +191,21 @@ std::pair<unsigned, unsigned> AMDGPUSubtarget::getEffectiveWavesPerEU(
       getOccupancyWithWorkGroupSizes(LDSBytes, FlatWorkGroupSizes).second};
   Default.first = std::min(Default.first, Default.second);
 
-  // Make sure requested minimum is less than requested maximum.
-  if (RequestedWavesPerEU.second &&
-      RequestedWavesPerEU.first > RequestedWavesPerEU.second)
+  // Make sure requested min is within the default range.
+  if (RequestedWavesPerEU.first < Default.first ||
+      RequestedWavesPerEU.first > Default.second)
     return Default;
 
-  // Make sure requested values do not violate subtarget's specifications and
-  // are compatible with values implied by minimum/maximum flat workgroup sizes.
-  if (RequestedWavesPerEU.first < Default.first ||
-      RequestedWavesPerEU.second > Default.second)
+  // When provided, make sure requested max is higher than min and does not
+  // violate target specification.
+  if (RequestedWavesPerEU.second &&
```

Comment:
I thought 0 was an illegal value

Reply:
I had in mind that the attributor sometimes called this method with a max of 0 to indicate "no maximum requested" but that appears to be incorrect, removed the 0-check and merged the two ifs.

Reply:
We did see a crash before and there was a PR trying to fix that. Now that code has gone.

Reply:
Thanks for the clarification, thought I was imagining stuff for a minute :)